# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/webrender_bindings/RenderD3D11TextureHost.h
# Commit: 7a990fa02246
# Full Hash: 7a990fa022462c702adacfac907a0fc178940b9d
# Author: Jean-Yves Avenard <jyavenard@mozilla.com>
# Date: 2020-10-21 09:51:28
# Regressor Bug: 1595994
# File Overlap Count: 1
# Description:
#   Bug 1595994 - P1D. Properly serialize display size when sending image over IPC. r=mattwoodrow
#   
#   The code always assumed that the size of the image with the Y plane dimensions, which, while often the case, isn't correct.
#   We remove the assertions that the display offset was always (0,0) and properly carry the actual data over IPC.
#   
# ==============================================================================

diff -r 4c7f671c7b69 -r 7a990fa02246 gfx/webrender_bindings/RenderD3D11TextureHost.h
--- a/gfx/webrender_bindings/RenderD3D11TextureHost.h	Tue Oct 20 23:29:30 2020 +0000
+++ b/gfx/webrender_bindings/RenderD3D11TextureHost.h	Tue Oct 20 23:30:04 2020 +0000
@@ -7,8 +7,8 @@
 #ifndef MOZILLA_GFX_RENDERD3D11TEXTUREHOST_H
 #define MOZILLA_GFX_RENDERD3D11TEXTUREHOST_H
 
+#include "GLTypes.h"
 #include "RenderTextureHost.h"
-#include "GLTypes.h"
 
 struct ID3D11Texture2D;
 struct IDXGIKeyedMutex;
@@ -77,7 +77,7 @@
 class RenderDXGIYCbCrTextureHost final : public RenderTextureHost {
  public:
   explicit RenderDXGIYCbCrTextureHost(WindowsHandle (&aHandles)[3],
-                                      gfx::IntSize aSize,
+                                      gfx::IntSize aSizeY,
                                       gfx::IntSize aSizeCbCr);
 
   wr::WrExternalImage Lock(uint8_t aChannelIndex, gl::GLContext* aGL,
@@ -109,7 +109,7 @@
   // The gl handles for Y, Cb and Cr data.
   GLuint mTextureHandles[3];
 
-  gfx::IntSize mSize;
+  gfx::IntSize mSizeY;
   gfx::IntSize mSizeCbCr;
 
   bool mLocked;
